### PR TITLE
Increasing fetchai test coverage

### PR DESCRIPTION
**Pull request by MissingNO57**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`scripts/generate_ipfs_hashes.py`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

File: aea/test_tools/test_contract.py [hunk 1]
```diff
@@ -143,20 +143,24 @@ def refill_from_faucet(
             raise ValueError("Balance not increased!")  # pragma: nocover
 
     @staticmethod
-    def sign_send_confirm_receipt_transaction(
-        tx: JSONLike, ledger_api: LedgerApi, crypto: Crypto, sleep_time: float = 1.0
+    def sign_send_confirm_receipt_multisig_transaction(
```
Comment (5A11) [inline aea/test_tools/test_contract.py:146]: Let's write a code comment (not docstring) for this function, that this function and `sign_send_confirm_receipt_transaction` below are there for backwards compatibility purposes. Next major release we can combine them (remove the one below and use this one.)

Comment (5A11) [reply aea/test_tools/test_contract.py:146]: Let's write a code comment (not docstring) for this function with the following format, explaining the situation so for the next major release we know what parts we can clean up.

`# BACKWARDS COMPATIBILITY: ....`

File: tests/test_packages/test_contracts/test_erc1155/test_contract.py [hunk 2]
```diff
@@ -773,38 +780,445 @@ def test_create_and_mint_and_balances(self):
...
+        ]
+
+    @pytest.mark.ledger
+    def test_get_create_batch_transaction_wrong_identifier(self):
+        """Test if get_create_batch_transaction with wrong api identifier fails."""
+
+        # Create mock ledger with unknown identifier
+        ledger_api = mock.Mock()
+        attrs = {"identifier": "dummy"}
+        ledger_api.configure_mock(**attrs)
```
Comment (5A11) [inline tests/test_packages/test_contracts/test_erc1155/test_contract.py:1046]: you can put these 3 lines in setup and reuse across the tests.

File: tests/test_packages/test_contracts/test_erc1155/test_contract.py [hunk 3]
```diff
@@ -773,38 +780,445 @@ def test_create_and_mint_and_balances(self):
...
+                agent_address="address",
+                token_ids=self.token_ids_a,
+            )
+
+    @pytest.mark.ledger
+    def test_get_hash_batch_not_same(self):
+        """Test if get_hash_batch returns ValueError when on-chain hash is not same as computed hash."""
+
+        ledger_api = mock.Mock()
+        ledger_api.identifier = "ethereum"
```
Comment (5A11) [inline tests/test_packages/test_contracts/test_erc1155/test_contract.py:1174]: Similar to the above, define in setup and reuse across.

**Comment by 5A11** [comment]
Please merge the latest develop and resolve any conflicts.

**Review by 5A11 — APPROVED** [review]
LGTM

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**